File: src/main/java/com/prediccion/apppredicciongm/gestion_inventario/movimiento/dto/response/KardexResponse.java
package com.prediccion.apppredicciongm.gestion_inventario.movimiento.dto.response;

import com.prediccion.apppredicciongm.enums.TipoMovimiento;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class KardexResponse {
    
    private Long kardexId;
    private Integer productoId;
    private String nombreProducto;
    private String categoriaProducto;
    private LocalDateTime fechaMovimiento;
    private TipoMovimiento tipoMovimiento;
    private String tipoDocumento;
    private String numeroDocumento;
    private Integer cantidad;
    private Integer saldoCantidad;
    private BigDecimal costoUnitario;
    private BigDecimal valorTotal;
    private String lote;
    private LocalDateTime fechaVencimiento;
    private Integer proveedorId;
    private String nombreProveedor;
    private String motivo;
    private String referencia;
    private Integer usuarioId;
    private String nombreUsuario;
    private String observaciones;
    private String ubicacion;
    private LocalDateTime fechaRegistro;
}

File: frontend/src/app/features/movimientos/model/KardexResponse.ts
import { Page } from "../../../shared/models/Page";

export interface KardexResponse {
    kardexId: number | null;
    productoId: number | null;
    nombreProducto: string | null;
    categoriaProducto: string | null;
    fechaMovimiento: string | null;
    tipoMovimiento: string | null;
    tipoDocumento: string | null;
    numeroDocumento: string | null;
    cantidad: number | null;
    saldoCantidad: number | null;
    costoUnitario: string | null;
    valorTotal: string | null;
    lote: string | null;
    fechaVencimiento: string | null;
    proveedorId: number | null;
    nombreProveedor: string | null;
    motivo: string | null;
    referencia: string | null;
    usuarioId: number | null;
    nombreUsuario: string | null;
    observaciones: string | null;
    ubicacion: string | null;
    fechaRegistro: string | null;
}
export interface PageKardexResponse extends Page<KardexResponse> { }